### ISSUE TITLE
Wiring packages does not work

### ISSUE THREAD

**Issue by languitar** [ets-labs/python-dependency-injector#307]
I have a dummy project with the following structure in which I am unable to get wiring running for packages. Modules work as expected.

Here's the structure:
```
❯ tree
.
├── poetry.lock
├── pyproject.toml
└── python_dependency_injector_test
    ├── dependencies.py
    ├── __init__.py
    ├── main.py
    └── package
        └── __init__.py
```

`dependencies.py`:
```python
from dependency_injector import containers, providers


class AppContainer(containers.DeclarativeContainer):
    dep = providers.Singleton(list, [1, 2])
```

`main.py`:
```python
from .dependencies import AppContainer
from . import package


if __name__ == "__main__":
    container = AppContainer()
    container.wire(packages=[package])
    package.test()
```

`package/__init__.py`:
```python
from dependency_injector.wiring import Provide

from ..dependencies import AppContainer


def test(
    dep=Provide[AppContainer.dep]
):
    print(f"dep: {dep}")
```

In my case, executing `poetry run python -m python_dependency_injector_test.main` always results in:
```
dep: <dependency_injector.wiring.Provide object at 0x7fb85161c070>
```

This happens on Python 3.8.6

**Comment by rmk135** [comment]
Hi @languitar. That's a bug. I have already fixed it locally. Will publish a new bugfix release in a couple of hours.

**Comment by rmk135** [comment]
Fixed in `4.0.6`. Took a little longer cause I ran into some issues with tests.

**Comment by rmk135** [comment]
Closing the issue. Please, comment / reopen if needed.
